query/src/main/java/org/infinispan/query/dsl/embedded/impl/EmbeddedLuceneQuery.java

```diff
@@ -115,6 +115,39 @@ public String toString() {
             '}';
    }
 
+   private IndexedQuery<T> getOrCreateIndexedQuery(boolean withHitCount) {
+      if (withHitCount) {
+         if (indexedQuery == null) {
+            indexedQuery = createQuery(true);
+         }
+         return indexedQuery;
+      } else {
+         if (indexedListQuery == null) {
+            indexedListQuery = createQuery(false);
+         }
+         return indexedListQuery;
+      }
+   }
+
+   private IndexedQuery<T> createQuery(boolean withHitCount) {
+      validateNamedParameters();
+      IndexedQuery<T> result = queryEngine.buildLuceneQuery(parsingResult, namedParameters, startOffset, maxResults, isLocal());
+
+      if (withHitCount) {
```

Comment:
Correct me if I am wrong, but don't want to apply the hit count accuracy setting even if the user doesn't return the hit count ?

Reply:
Yes we do. I think that in case the hit count is not returned (e.g., #list) the hit count accuracy is limited to 1. In this way we're saying to Hibernate Search and in turn to Lucene that we don't need to compute this value.